services/src/main/java/io/scalecube/services/ServiceInfo.java

```diff
@@ -0,0 +1,79 @@
+package io.scalecube.services;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Helper class used to register service with tags as metadata in the scalecube cluster.
+ * parsing from service info to json and back.
+ */
+public class ServiceInfo {
+
+  @JsonIgnore
+  private static final ObjectMapper mapper = new ObjectMapper();
```

Comment:
oh... it didn't show it right... it is not a delimiter:

example is following - you have service with name myCoolService and also you have on this service a tag with name myCoolTag with value 42, then it means that in matadata you are going to have following property:

myCoolService:tag:myCoolTag=42

so since you know that your service name is "myCoolService" and also service names are not allowed to have ":" symbol inside you can safely search over metadata properties for the ones which starts with prefix "myCoolService:tag:", then you remove this prefix to get the value of tag key which is "myCoolTag" and value of property is actually value of the tag.